packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -25,8 +25,11 @@ class Firestore {
             ? platform.FirestorePlatform.instanceFor(app: app)
             : platform.FirestorePlatform.instance;
 
+  // Cache single instance
+  static Firestore _instance;
+
   /// Gets the instance of Firestore for the default Firebase app.
-  static Firestore get instance => Firestore();
+  static Firestore get instance => _instance ??= Firestore();
```

Comment:
@slightfoot 

This seems redundant for me because `Firestore()` is practically singleton.

`static Firestore get instance` calls these:

1. https://github.com/FirebaseExtended/flutterfire/blob/5e380cf85fde7eefa91d520bdb3ae0bfeb2d1282/packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart#L26
2. https://github.com/FirebaseExtended/flutterfire/blob/5e380cf85fde7eefa91d520bdb3ae0bfeb2d1282/packages/cloud_firestore/cloud_firestore_platform_interface/lib/cloud_firestore_platform_interface.dart#L61-L66
    - This is singleton